addons/website_payment/models/payment_transaction.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, fields, models
+
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    is_donation = fields.Boolean(string="Is donation", help="Is the payment a donation")
+
+    def _finalize_post_processing(self):
+        for tx in self.filtered('is_donation'):
+            tx._send_donation_email()
+        super()._finalize_post_processing()
+
+    def _send_donation_email(self, notif=False, comment=None, recipient_email=None):
+        self.ensure_one()
+
+        subject = _('A donation has been made on your website') if notif else _('Donation confirmation')
+        sentence = "" if notif else _('Thank you for your donation. Your generosity is appreciated!')
+        recordName = _('Donation notification') if notif else _('Donation')
+        recipientEmail = recipient_email if notif else self.partner_email
+        modelDescription = _('donation notification') if notif else _('donation')
+
+        if notif or self.state == 'done':
+            # send donation mail
+            body = self.env.ref('website_payment.donation_mail')._render({
+                'fullname': self.partner_name,
+                'email': self.partner_email,
+                'date': self.create_date,
+                'amount': self.amount,
+                'currency': self.currency_id,
+                'payment_method': self.provider,
+                'payment_id': self.reference,
+                'sentence': sentence,
+                'notif': notif,
+                'comment': comment,
+            }, engine='ir.qweb', minimal_qcontext=True)
+            message = self.env['mail.message'].sudo().new({
+                'body': body,
+                'record_name': recordName,
+            })
+            body_html = self.env.ref('mail.mail_notification_light')._render({
+                'message': message,
+                'model_description': modelDescription,
+                'company': self.company_id,
+            }, engine='ir.qweb', minimal_qcontext=True)
+            body_html = self.env['mail.render.mixin']._replace_local_links(body_html)
+
```

Comment:
why reinvent the `mail.template` logic here?

Reply:
I cannot find a clean way to inject non-model fields in the body rendering of the `mail.template` mechanism.
(e.g. the `comment` field is a user input that never gets stored in a model, it just gets included in the email)
I could either:
- update the template's call to `_render_field()` to also specify an `add_context` and add a parameter to `send_mail` to make it reach that point
- or pre-render the `body_html` field then provide it in the `email_values` of `send_mail`, but that sounds like defeating the purpose of a template

Reply:
I tried adding a `donation_comment` field to the payment transaction, but even this never led to the expected emails because the texts in the wrapping template were always obtained from the model's display name ("Payment Transaction") and the record's display name ("tx-2021...") instead of the required "Donation"/"Donation notification".
Also the template body had to be duplicated because the `body_html` field is not rendered through QWeb - thus not making it possible to use a `t-call`.

Reply:
After further discussion with @JKE-be, using the partial `mail_template` solution.
Added a "Donation"/"Donation notification" title inside the mail content since it cannot be put inside the template header itself.